tests/utils/atomWithDefault.test.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { Fragment, Suspense } from 'react'
+import { fireEvent, render } from '@testing-library/react'
+import { Provider as ProviderOrig, atom, useAtom } from '../../src/index'
+import { atomWithDefault } from '../../src/utils'
+
+const Provider = process.env.PROVIDER_LESS_MODE ? Fragment : ProviderOrig
+
+it('simple sync get default', async () => {
+  const count1Atom = atom(1)
+  const count2Atom = atomWithDefault((get) => get(count1Atom) * 2)
+
+  const Counter: React.FC = () => {
+    const [count1, setCount1] = useAtom(count1Atom)
+    const [count2, setCount2] = useAtom(count2Atom)
+    return (
+      <>
+        <div>
+          count1: {count1}, count2: {count2}
+        </div>
+        <button onClick={() => setCount1((c) => c + 1)}>button1</button>
+        <button onClick={() => setCount2((c) => c + 1)}>button2</button>
+      </>
+    )
+  }
+
+  const { findByText, getByText } = render(
+    <Provider>
+      <Counter />
+    </Provider>
+  )
+
+  await findByText('count1: 1, count2: 2')
+
+  fireEvent.click(getByText('button1'))
+  await findByText('count1: 2, count2: 4')
+
+  fireEvent.click(getByText('button2'))
+  await findByText('count1: 2, count2: 5')
+
+  fireEvent.click(getByText('button1'))
+  await findByText('count1: 3, count2: 5')
+})
+
+it('simple async get default', async () => {
+  const count1Atom = atom(1)
+  const count2Atom = atomWithDefault(async (get) => {
+    await new Promise((r) => setTimeout(r, 10))
+    return get(count1Atom) * 2
```

Comment:
Maybe a test missing with async `set`

Reply:
Hadn't thought about it. Would it be possible for you to add such a test?

Reply:
Sure, I'll do it

Reply:
I first thought it's a neat idea, but then this doesn't work... unfortunately. async just works more or less accidentally. the `write` may not usually write self (though technically possible except for types), so this is counter-idiomatic too.
Thanks for your work anyway. I got an idea to improve the impl. (I did something weird, to try something else.)